Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -338,48 +387,84 @@ -(void)updateDisplayOrientedPose:(ARCamera*)camera {
 }
 
 - (void)session:(ARSession *)__unused session didAddAnchors:(nonnull NSArray<__kindof ARAnchor *> *)anchors {
-    if (!planeDetectionEnabled) {
-        return;
+    if (planeDetectionEnabled) {   
+        [self LockPlanes];
+        for (ARAnchor* newAnchor : anchors) {
+            if ([newAnchor isKindOfClass:[ARPlaneAnchor class]]) {
+                updatedPlanes.insert((ARPlaneAnchor*)newAnchor);
+            }
+        }
+        
+        [self UnlockPlanes];
     }
-
-    [self LockPlanes];
-    for (ARAnchor* newAnchor : anchors) {
-        if ([newAnchor isKindOfClass:[ARPlaneAnchor class]]) {
-            updatedPlanes.insert((ARPlaneAnchor*)newAnchor);
+    
+    if (meshDetectionEnabled) {
+        [self LockMeshes];
+        for (ARAnchor* newAnchor : anchors) {
+            if (@available(iOS 13.4, *)) {
```

Comment:
Might as well put this in the meshDetectionEnabled if statement, so we don't need to grab the lock if the API is unavailable.